homeassistant/components/google_assistant/__init__.py

```diff
@@ -36,17 +47,58 @@
             vol.Required(CONF_ACCESS_TOKEN): cv.string,
             vol.Optional(CONF_EXPOSE_BY_DEFAULT): cv.boolean,
             vol.Optional(CONF_EXPOSED_DOMAINS): cv.ensure_list,
+            vol.Optional(CONF_AGENT_USER_ID,
+                         default=DEFAULT_AGENT_USER_ID): cv.string,
+            vol.Optional(CONF_HOMEGRAPH_API_KEY): cv.string
         }
     },
     extra=vol.ALLOW_EXTRA)
 
 
+@bind_hass
+def request_sync(hass):
+    """Request sync."""
+    hass.services.call(DOMAIN, SERVICE_REQUEST_SYNC)
+
+
 @asyncio.coroutine
 def async_setup(hass: HomeAssistant, yaml_config: Dict[str, Any]):
     """Activate Google Actions component."""
     config = yaml_config.get(DOMAIN, {})
-
+    agent_user_id = config.get(CONF_AGENT_USER_ID)
+    api_key = config.get(CONF_HOMEGRAPH_API_KEY)
+    descriptions = yield from hass.async_add_job(
+        conf_util.load_yaml_config_file, os.path.join(
+            os.path.dirname(__file__), 'services.yaml')
+    )
     hass.http.register_view(GoogleAssistantAuthView(hass, config))
     hass.http.register_view(GoogleAssistantView(hass, config))
 
+    @asyncio.coroutine
+    def request_sync_service_handler(call):
```

Comment:
Need to move this out of the `async_setup` function 😁

Reply:
Hey @philk - confused by this comment. I have followed examples from automation, light inits. They all seem to define the service handler inside async_setup

Reply:
Stylistically it looks weird to me but I looked through a few other components and that just seems to be the way to do it for hass/in python so nevermind, it works and matches so LGTM